docs/executor-custom.md

```diff
@@ -35,39 +44,80 @@ type Runner interface {
+
+Example runner is defined in our template - so if you'll use it only thing which need to be done is implementing Run method (you can rename `ExampleRunner` to whatever business name describing your testing framework)
+
+Runner can get data from different sources - for now we're supporting:
+
+- string content (e.g. Postman JSON file)
+- URI - content stored on webserver
+- Git File - file storeg in Git repo in given path
+- Git Dir - whole git repo, or git subdirectory (we'll do spatial checkout to save traffic in case of monorepos)
+
+all possible test definitions are already created and mounted as Kubernetes `Volume` before executor starts its work. You can get directory path from `RUNNER_DATADIR` environment variable. 
 
 ```go
-// ExampleRunner for template - change me to some valid runner
+
+// TODO: change me to some valid name
 type ExampleRunner struct {
 }
 
 func (r *ExampleRunner) Run(execution testkube.Execution) (testkube.ExecutionResult, error) {
- return testkube.ExecutionResult{
-  Status: testkube.StatusPtr(testkube.SUCCESS_ExecutionStatus),
-  Output: "exmaple test output",
- }, nil
+ 
+  // execution.Content could have git repo data
+  // we're also passing content files/directories as mounted volume in directory
+  path := os.Getenv("RUNNER_DATADIR")
+
+  // e.g. Cypress test is stored in Git repo so Testkube will checkout it automatically 
+  // and allow you to use it easily
+
+  uri := execution.Content.Data
+  resp, err := http.Get(uri)
+  if err != nil {
+    return result, err
+  }
+  defer resp.Body.Close()
+
+  b, err := ioutil.ReadAll(resp.Body)
+  if err != nil {
+    return result, err
+  }
+
+  // if get is successful return success result
+  if resp.StatusCode == 200 {
+    return testkube.ExecutionResult{
+      Status: testkube.ExecutionStatusSuccess,
+      Output: string(b),
+    }, nil
+  }
+
+  // else we'll return error to simplify example
+  err = fmt.Errorf("invalid status code %d, (uri:%s)", resp.StatusCode, uri)
+  return result.Err(err), nil
 }
+
 ```
 
 Runner need to return `ExecutionResult` or `error` (in case of runner can't run tests), ExecutionResult
 could have different statuses (look at OpenAPI spec for details) - we'll focus on `success` and `error`
 
-Let's assume that user will create test which content will be simply URI to test.
+Additionally we could want to parse test framework test parts (e.g. different test steps) we should make some 
+mapping between particular testing framework and Testkube itself (keep in mind that we've skipped those details here to simplify example).
+
+If we're running any testing framework binary it's good to wrap its output 
+
+Example of [mapping in Testkube Postman Executor](https://github.com/kubeshop/testkube-executor-postman/blob/main/pkg/runner/newman/newman.go#L60), which using [Postman to Testkube Mapper](https://github.com/kubeshop/testkube-executor-postman/blob/1b95fd85e5b73e9a243fbff59d5e96c27d0f69c5/pkg/runner/newman/mapper.go#L9)
```

Comment:
Here is an example of [mapping in Testkube Postman Executor](https://github.com/kubeshop/testkube-executor-postman/blob/main/pkg/runner/newman/newman.go#L60), which is using [Postman to Testkube Mapper](https://github.com/kubeshop/testkube-executor-postman/blob/1b95fd85e5b73e9a243fbff59d5e96c27d0f69c5/pkg/runner/newman/mapper.go#L9).